### PR TITLE
Add checkperf to cmake

### PR DESCRIPTION
**Pull request by jkeiser**
You can run it like:

```bash
cmake .
make parse perfdiff
ctest -R checkperf
```

We now check for performance degradation in all implementations (haswell, westmere, fallback).

### CODE REVIEW

**Comment by jkeiser** [comment]
My plan here is to get this to the point where checkperf *runs* in cmake, and then turn it off except in 1 or 2 places. We don't need to increase the number of chances of a false positive checkperf right now.

**Comment by jkeiser** [comment]
Hrm. haswell-clang6 failed the perf test twice in a row. One issue with the way I did this was, it doesn't actually use the same flags when it compiles master.

I'll see if I can't fix that before moving on ... it feels wrong to leave it incomplete.

**Comment by jkeiser** [comment]
If this passes, I'm going to try adding westmere, unthreaded, and fallback checkperf tests. Note that even if those work, we're going to be increasing the volatility of our CI (every time we add a checkperf, volatility is going to increase).

**Comment by jkeiser** [comment]
OK, that passed! Here goes nothing ...

File: .drone.yml [hunk 1]
```diff
@@ -462,7 +462,7 @@ steps:
     - mkdir build && cd build
     - cmake $CMAKE_FLAGS ..
     - make -j
-    - ASAN_OPTIONS="detect_leaks=0" ctest -j --output-on-failure
+    - ASAN_OPTIONS="detect_leaks=0" ctest -j4 --output-on-failure -E checkperf
```
Comment (lemire) [inline .drone.yml:465]: Why would we ever want to run checkperf on sanitized code?

Comment (jkeiser) [reply .drone.yml:465]: @lemire -E is exclude :)

Comment (lemire) [reply .drone.yml:465]: Ok. Then I am fine.

**Pull request merged**